android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,240 @@
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            registerNotificationChannels(applicationContext)
+        }
+        Logger.tag(TAG).debug { "Breez node service created." }
+    }
+
+    // =========================================================== //
+    //                      SERVICE LIFECYCLE                      //
+    // =========================================================== //
+
+    override fun onBind(intent: Intent?): IBinder {
+        Logger.tag(TAG).debug { "Binding Breez node service from intent=$intent" }
+        paymentReceivedInBackground.clear()
+        stopForeground(STOP_FOREGROUND_REMOVE)
+        dismissForegroundServiceNotification(applicationContext)
+        return binder
+    }
+
+    override fun onUnbind(intent: Intent?): Boolean {
+        return false
+    }
+
+    private val shutdownHandler = Handler(Looper.getMainLooper())
+    private val shutdownRunnable: Runnable = Runnable {
+        Logger.tag(TAG).debug { "Reached scheduled shutdown..." }
+        if (paymentReceivedInBackground.isEmpty()) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            stopForeground(STOP_FOREGROUND_DETACH)
+        }
+        shutdown()
+    }
+
+    /** Shutdown the node, close connections and stop the service */
+    private fun shutdown() {
+        Logger.tag(TAG).info { "Shutting down Breez node service" }
+        stopSelf()
+        _state.postValue(NodeServiceState.Off)
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        Logger.tag(TAG).debug {
+            "Start Breez node service from intent [ intent=$intent, flag=$flags, startId=$startId ]"
+        }
+        intent.getRemoteMessage()?.let {
+            if (it.data["notification_type"] == "payment_received") {
+                val paymentHash = it.data["payment_hash"]
+                val clickAction = it.data["click_action"]
+                paymentHash?.let {
+                    startBusiness(paymentHash, clickAction)
+                }
+            }
+        }
+
+        shutdownHandler.removeCallbacksAndMessages(null)
+        shutdownHandler.postDelayed(shutdownRunnable, 120 * 1000L) // push back shutdown by 120s
+        return START_NOT_STICKY
+    }
+
+    private fun startBusiness(
+        paymentHash: String,
+        clickAction: String?,
+    ) {
+        val notification = notifyForegroundService(applicationContext)
+        when {
+            breezSDK != null && _state.value is NodeServiceState.Running -> {
```

Comment:
This whole state management seems complicated. Why do we need it (NodeState)?
What's wrong in simple callling connectSDK (which already handles a singleton static instance) ?